src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -308,37 +363,33 @@ public static object CoerceValue(ISchema schema, IGraphType type, IValue input,
 
                 if (input is ListValue list)
                 {
-                    return list.Values
-                        .Select(item => CoerceValue(schema, listItemType, item, variables))
-                        .ToList();
+                    return new ArgumentValue(list.Values
+                        .Select(item => CoerceValue(schema, listItemType, item, variables).Value)
+                        .ToList(), ArgumentSource.Literal);
                 }
                 else
                 {
-                    return new[] { CoerceValue(schema, listItemType, input, variables) };
+                    return new ArgumentValue(new[] { CoerceValue(schema, listItemType, input, variables).Value }, ArgumentSource.Literal);
                 }
             }
 
-            if (type is IObjectGraphType || type is IInputObjectGraphType)
```

Comment:
What about `type is IObjectGraphType` condition?

Reply:
`IObjectGraphType` is invalid for input types so it is not applicable.

Reply:
WOW 🤦 

For today I'm done, I have looked at almost all the files, tomorrow I will finish.

Reply:
Thanks!! 👍